RobotCode/snobot2018/src/org/snobot/elevator/SnobotElevator.java

```diff
@@ -48,30 +60,41 @@ private void caluculateHeight()
     {
 
         double encoderPosition = mElevatorEncoder.getDistance();
-        mHeight = encoderPosition;
+        mActualHeight = encoderPosition;
     }
 
 
     @Override
     public void control()
     {
-        setMotorSpeed(mElevatorSpeed);
 
+        if (mGotoHeight)
+        {
+            gotoHeight();
+        }
+        else
+        {
+            setMotorSpeed(mJoystickSpeed);
+        }
     }
 
     @Override
     public void stop()
     {
+        mGotoHeight = false;
         setMotorSpeed(0);
-
     }
 
     @Override
     public void update()
     {
         caluculateHeight();
-        mElevatorSpeed = mJoystick.getElevatorSpeed();
+        mJoystickSpeed = mJoystick.getElevatorSpeed();
 
+        if (mJoystickSpeed != 0)
```

Comment:
Either here or in the IO you should put a deadband around this.  Xbox controllers are never 0, you might want to shut this off if it is `-.01 < x < .01` or something